### DISCUSSION TITLE
Getting 401

### DISCUSSION THREAD

**Discussion by guitaripod** [ankitpokhrel/jira-cli#459] [Q&A]
I'm only getting 401 or 302 depending on if I give it the host only or more.

I've read that the username might be what is before the @ in the email account. That doesn't work either.

Could someone verify if the readme instructions are up to date? Jira might've changed things.

Also, my name is with é when I'm tagged on Jira. Could that be it?
<img width="150" alt="CleanShot 2022-09-05 at 16 23 05@2x" src="https://user-images.githubusercontent.com/47460844/188459878-60277988-dc1d-4b98-ae7a-08efeff7de36.png">

```
MaxBook ~ $ jira init --debug
? Installation type: Local
? Link to Jira server: https://company.atlassian.net
? Login username: me@company.com
⠸ Verifying login details...

REQUEST DETAILS
------------------------------------------------------------

GET /rest/api/2/myself HTTP/1.1
Host: company.atlassian.net
Authorization: Bearer bearer



RESPONSE DETAILS
------------------------------------------------------------

HTTP/1.1 401 Unauthorized
Transfer-Encoding: chunked
Atl-Traceid: string
Cache-Control: no-transform
Content-Type: application/xml;charset=UTF-8
Date: datestring
Expect-Ct: report-uri="https://web-security-reports.services.atlassian.com/expect-ct-report/atlassian-proxy", max-age=86400
Nel: {"failure_fraction": 0.001, "include_subdomains": true, "max_age": 600, "report_to": "endpoint-1"}
Report-To: {"endpoints": [{"url": "https://string.cloudfront.net"}], "group": "endpoint-1", "include_subdomains": true, "max_age": 600}
Server: globaledge-envoy
Set-Cookie: atlassian.xsrf.token=string; path=/; SameSite=None; Secure
Strict-Transport-Security: max-age=63072000; preload
Timing-Allow-Origin: *
Vary: Accept
Www-Authenticate: OAuth realm="https.string.atlassian.net"
X-Arequestid: string
X-Content-Type-Options: nosniff
X-Envoy-Upstream-Service-Time: 220
X-Xss-Protection: 1; mode=block


✗ Received unexpected response '401 Unauthorized' from jira. Please try again.
```
![image](https://user-images.githubusercontent.com/47460844/188459320-c62b1ee2-a693-4392-b5bd-964491df6dd7.png)

**Comment by ankitpokhrel** [comment]
Hi @marcusziade, to confirm again is this a Local or Cloud (atlassian.net) installation?

**Reply by guitaripod** [reply]
Hello, thanks for the reply. It's local. It's the second line in the snippet I posted

**Reply by guitaripod** [reply]
I'm already apologizing if I come on too strong. I've been banging my head against the wall for hours.

**Reply by guitaripod** [reply]
Also, here's proof I know what I'm doing somewhat. The key is accessed with certain queries 
![CleanShot 2022-09-05 at 16 25 49](https://user-images.githubusercontent.com/47460844/188460320-614e7367-9159-4a6a-966c-a232a0bc8d07.png)

**Reply by ankitpokhrel** [reply]
No worries, If this is a local installation.

**Option 1: Basic Auth**
- Set `export JIRA_AUTH_TYPE=basic` or unset this variable entirely.
- Set `export JIRA_API_TOKEN=<your-login-password>`
- Select installation type as `Local` during `jira init`
- use your username

**Option 2: Personal Access Token**
- Set `export JIRA_AUTH_TYPE=bearer` or unset this variable entirely.
- Set `export JIRA_API_TOKEN=<your-pat-token>`
- Select installation type as `Local` during `jira init`
- use your username

**Reply by guitaripod** [reply]
I'm doing Option 2:

I've done exactly this. I've tried using my full email and just marcus.ziade and what Jira posts when I'm tagged in the comments, but no luck.

**Reply by guitaripod** [reply]
Check [this](https://community.atlassian.com/t5/Jira-questions/How-to-find-username-of-my-jira-account/qaq-p/969298)

Could Jira have made some change?

Have you tried the init yourself lately? Do you have tests written for it?

**Reply by guitaripod** [reply]
I've tried the number it shows in the HTML, but that doesn't work either. I'm getting accessed on the key with `marcus.ziade`

**Reply by ankitpokhrel** [reply]
Guessing from the snapshot you shared above, the page seem to be for API tokens not the Personal Access Tokens (PAT). Those are different things. For PAT in Jira, select your profile picture at the top right of the screen, then choose Profile. Once you access your profile, select Personal Access Tokens in the left-hand menu.

**Reply by guitaripod** [reply]
Do I need to have access to them? Could it be my admins have disabled it?

**Reply by ankitpokhrel** [reply]
Yes, it might be that it is disabled in your installation. In that case you can either ask your Admin to enable it or try with Basic Auth.

**Reply by guitaripod** [reply]
How do I go about basic auth? I don't think that works as I sign in to Jira through Google authentication in the browser. I can't sign in with the email, password form they have.

**Reply by ankitpokhrel** [reply]
hmm...I don't know what I can do here right now. If you can't use PAT or get username/pwd setup for your account then I think you'll need to wait until cookie based authentication is implemented.

**Reply by guitaripod** [reply]
@ankitpokhrel What's the status on Google 2fa login?
